ScreenShareIndicatorFrame.c

```diff
@@ -0,0 +1,368 @@
+	struct FindWindowData findWindowData = { 
+		trackedWindow, 
+		FALSE 
+	};
+	
+	EnumWindows( findWindowHandle, (LPARAM) &findWindowData );
+	if( !findWindowData.found ) {
+		return EXIT_FAILURE;
+	}
+
+    WNDCLASSA wc = { 
+		CS_OWNDC, 
+		(WNDPROC) trackWindowWndProc, 
+		0, 
+		0, 
+		GetModuleHandleA( NULL ), 
+		NULL, 
+		NULL, 
+		NULL, 
+		NULL, 
+		WINDOW_CLASS_NAME 
+	};
+
+    RegisterClassA( &wc );
+
+    HWND hwnd = CreateWindowExA( WS_EX_LAYERED | WS_EX_TRANSPARENT | WS_EX_TOOLWINDOW, wc.lpszClassName, 
+		NULL, WS_BORDER,  CW_USEDEFAULT, CW_USEDEFAULT, CW_USEDEFAULT, CW_USEDEFAULT, NULL, NULL, 
+		GetModuleHandleA( NULL ), 0 );
+
+	COLORREF transparency_key = RGB( 0, 255, 255 );
+	struct WindowData windowData = { 
+		trackedWindow, 
+		CreateSolidBrush( transparency_key ), 
+		CreateSolidBrush( FRAME_COLOR ) 
+	};
+
+	SetWindowLongPtrA( hwnd, GWLP_USERDATA, (LONG_PTR)&windowData );
+	SetWindowLongA(hwnd, GWL_STYLE, WS_VISIBLE );
+	SetLayeredWindowAttributes(hwnd, transparency_key, 0, LWA_COLORKEY);
+	SetWindowLongPtrA( hwnd, GWLP_HWNDPARENT, (LONG_PTR) trackedWindow );
+	trackWindowTimerProc( hwnd, 0, 0, 0 );
+
+
+	MSG msg = { NULL };
+	while (GetMessage(&msg, NULL, 0, 0)) {
+		TranslateMessage(&msg);
+		DispatchMessage(&msg);
+	}
+
+	DeleteObject( windowData.foreground );
+	DeleteObject( windowData.background );
+	DestroyWindow( hwnd );
+	UnregisterClassA( wc.lpszClassName, GetModuleHandleA( NULL ) );
+
+	return EXIT_SUCCESS;
+}
+
+
+
+struct FindScreenData {
+	int x;
+    int y;
+	BOOL found;
+	RECT bounds;
+};
+
+
+static BOOL CALLBACK findScreen( HMONITOR monitor, HDC dc, LPRECT rect, LPARAM lparam ) {	
+	struct FindScreenData* findScreenData = (struct FindScreenData*) lparam;
+
+	if( rect->left == findScreenData->x && rect->top == findScreenData->y ) {
+		findScreenData->found = TRUE;
+		findScreenData->bounds = *rect;
+		return FALSE;
+	}
+
+	return TRUE;
+}
+
+
+struct ScreenData {
+	HDC deviceContext;
+	HBRUSH background;	
+	HBRUSH foreground;	
+};
+
+
+static LRESULT CALLBACK trackScreenWndProc( HWND hwnd, UINT message, WPARAM wparam, LPARAM lparam) {
+	struct ScreenData* screenData = (struct ScreenData*) GetWindowLongPtrA( hwnd, GWLP_USERDATA );
+	switch( message ) {
+		case WM_CLOSE: {
+			PostQuitMessage( 0 );
+		} break;
+		case WM_PAINT: {
+			RECT r;
+			GetClientRect( hwnd, &r );
+			FillRect( screenData->deviceContext, &r, screenData->foreground );
+			InflateRect( &r, -5, -5 );
+			r.bottom -= 1;
+			FillRect( screenData->deviceContext, &r, screenData->background );
+		} break;
+	}
+    return DefWindowProc( hwnd, message, wparam, lparam);
+}
+
+
+static int trackScreen( int x, int y ) {
+	struct FindScreenData findScreenData = { 
+		x, y,
+		FALSE 
+	};
+	
+	EnumDisplayMonitors( NULL, NULL, findScreen, (LPARAM) &findScreenData );
+	if( !findScreenData.found ) {
+		return EXIT_FAILURE;
+	}
+
+    WNDCLASSA wc = { 
+		CS_OWNDC | CS_HREDRAW | CS_VREDRAW, 
+		(WNDPROC) trackScreenWndProc, 
```

Comment:
What if the tracking window is closed? calls `PostQuitMessage( 0 );` and the process terminates?

Reply:
Yes, if the window is closed, it gets the WM_CLOSE message on line 211, and does PostQuitMessage. This will make the loop on line 271 exit (GetMessage returns false when WM_QUIT is received, and then a controlled exit happens.